storage/filestore.go

```diff
@@ -221,7 +221,7 @@ func (f *FilesystemStore) Set(name string, meta []byte) error {
 	os.RemoveAll(fp)
 
 	// Write the file to disk
-	if err = ioutil.WriteFile(fp, meta, f.perms); err != nil {
+	if err = ioutil.WriteFile(fp, meta, f.perms&notary.PermExecutableMask); err != nil {
```

Comment:
Thanks to the GFW I can't get access to the site  https://gas.css.bio/#docker/notary .

Do you think it make sense that removing the function `NewFilesystemStore()` because I find it was only  used to join the path , so we can use `NewFileStore()` to replace it.

Besides, if we decide to drop the concept of public files, do you think it also make sense that re-define the function  `NewFileStore(baseDir, fileExt string, perms os.FileMode)`   to just `NewFileStore(baseDir, fileExt string)` and set the permission to `0700/0600` wherever it needs (the only place is the filestore.Set()).